### PR TITLE
[RELAND] Remove some unnecessary <iostream> includes from headers

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108150
* #108149

In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305

### CODE REVIEW

File: .github/ci_commit_pins/xla.txt [hunk 1]
```diff
@@ -1 +1 @@
-709c75a04d9b35d83a9509e1534a8aa2046b8912
+c3c16ccac41cb2db6ba88fb31342f4af62c7e15a
```
Comment (albanD) [inline .github/ci_commit_pins/xla.txt:1]: Should not be there?

Comment (peterbell10) [reply .github/ci_commit_pins/xla.txt:1]: This is for https://github.com/pytorch/xla/pull/5434, xla was relying on indirectly including iostream through our headers.

**Review by albanD — APPROVED** [review]
Ok!

**Comment by albanD** [comment]
@peterbell10 could you fix the merge conflicts please?

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**